data/core/units/monsters/Fungoid_Spore.cfg

```diff
@@ -0,0 +1,126 @@
+#textdomain wesnoth-units
+
+[unit_type]
+    id=Fungoid Spore
+    name= _ "Fungoid Spore"
+    race=monster
+    image=units/monsters/orb-small.png
+    hitpoints=30
+    movement_type=smallfoot
+    movement=4
+    [resistance]
+        blade=100
+        pierce=100
+        impact=90
+        fire=110
+        cold=90
+        arcane=120
+    [/resistance]
+    experience=18
+    level=1
+    alignment=chaotic
+    advances_to=Fungoid Stalk
+    cost=13
+    usage=mixed fighter
+    description= _ "This thing is impossible to describe, no one has seen anything like it before."
+    die_sound=wail-long.wav
+
+    [abilities]
+        {ABILITY_REGENERATES}
+        [dummy]
+            id=miasma
+            name= _ "miasma"
+            description=_ "This unit passively harms those not of its side at a distance and rate equal to its level."
+        [/dummy]
+    [/abilities]
+    # Miasma ability.
+    [event]
+        id=Fungoid Miasma
```

Comment:
This appears to be duplicated in the three units. Couldn't it be made into a macro?

Reply:
It probably could be a macro but is too immature to become a _core_ macro. This ability could end up being dropped or modified and renamed. For example, right now it is conceptualized as a near neutrally buoyant noxious gas, so it floats everywhere and ignores terrain defenses. But it could end up considering terrain defense and then also need a new name and description.

Reply:
I think reimplementing it in Lua would be preferable to a macro, but a macro is preferable to repeating the exact same code in 3 places.

Reply:
I am game to try rewriting this in Lua if that is your recommendation.